web/lib/supabase/groups.ts

```diff
@@ -92,3 +92,32 @@ export async function getGroupsWhereUserIsMember(userId: string) {
 
   return groupThings.data
 }
+
+export async function getNonPublicGroupsWhereUserHasRole(userId: string) {
+  const groupThings = await run(
+    db
+      .from('group_role')
+      .select('group_data')
+      .eq('member_id', userId)
+      .or(
+        'group_data->>privacyStatus.eq.private,group_data->>privacyStatus.eq.restricted'
+      )
+      .or('role.eq.admin,role.eq.moderator')
```

Comment:
I could be wrong, but shouldn't these be `.and` queries?

Reply:
no the whole thing is an or statement between the two conditions